modelon/impact/client/entities/case.py

```diff
@@ -744,9 +762,21 @@ def sync(self) -> None:
             case.sync()
 
         """
-        self._info = self._sal.experiment.case_put(
-            self._workspace_id, self._exp_id, self._case_id, self._info
-        )
+        info = self._info
+        server_info = self._get_info()
+        skip_keys = [
+            "fmu_id",
+            "analysis_function",
+            "structural_parametrization",
+            "fmu_base_parametrization",
+            "run_info",
+            "id",
+        ]
+        info["run_info"] = server_info.get("run_info", {})
+        if not _compare_nested_dicts(info, server_info, skip_keys):
```

Comment:
I think it would be simpler to skip this and always call case_put. The performance is not critical here and it adds a lot of code. In the future we will probably have a PATCH method also would change the code in the future so adding a more complicated solution is probably not worth it.

Reply:
I feel like we should avoid all that complexity and just add the PATCH end-point into API.

Other aspect I'm not sure about is if it's really worth it to always do `sync()` (i.e., PUT/PATCH meta) before each execute unconditionally. We could instead introduce some flag `_is_updated` that is set in each property setter method and only PUT/PATCH if that flag is set. Same flag may need checking in "get_info" to avoid override. This is relevant if:
- User sets, e.g., a label in a script and calls `get_info` (no cache) that overrides the label
- A user is switching between UI and notebook, e.g., setting label via UI which can get override from script